Lib/test/test_clinic.py

```diff
@@ -1534,6 +1577,35 @@ def test_cli_converters(self):
                     f"expected converter {converter!r}, got {line!r}"
                 )
 
+    def test_cli_fail_converters_and_filename(self):
+        out = self.expect_failure("--converters", "test.c")
+        msg = (
+            "Usage error: can't specify --converters "
+            "and a filename at the same time"
+        )
+        self.assertIn(msg, out)
+
+    def test_cli_fail_no_filename(self):
+        out = self.expect_failure()
+        self.assertIn("usage: clinic.py", out)
+
+    def test_cli_fail_output_and_multiple_files(self):
+        out = self.expect_failure("-o", "out.c", "input.c", "moreinput.c")
+        msg = "Usage error: can't use -o with multiple filenames"
+        self.assertIn(msg, out)
+
+    def test_cli_fail_filename_or_output_and_make(self):
+        for opts in ("-o", "out.c"), ("filename.c",):
```

Comment:
Style nit:

```suggestion
        for opts in [("-o", "out.c"), ("filename.c",)]:
```

For me, this is much more readable.